### PR TITLE
Add verification for Network parameter DHCLIENT_SET_HOSTNAME

### PR DESCRIPTION
**Pull request by jknphy**
Verify that **DHCLIENT_SET_HOSTNAME** preloaded from control file does not overwrite the customer selection and check that hostname in the installed system is the right one according https://github.com/openSUSE/linuxrc/blob/master/linuxrc_hostname.md. Hostname in the form linux-xxxx is only reached if the installer really disables dhcp (not just the verification that the setting is properly set) given that the test include a fixed hostname **NICTYPE_USER_OPTIONS=hostname=myguest** which is taken into account only when is using dhcp.

- Related ticket: https://progress.opensuse.org/issues/25724
- Needles: https://gitlab.suse.de/openqa/os-autoinst-needles-sles/merge_requests/710
- Verification runs:
  - [sle-15-Installer-DVD-x86_64-Build459.1-network_configuration@64bit](http://dhcp227/tests/605)
  - [sle-12-SP4-Server-DVD-x86_64-Build0230-network_configuration@64bit](http://dhcp227/tests/609)

### CODE REVIEW

**Comment by SergioAtSUSE** [comment]
Sorry, I missed the WIP

**Comment by jknphy** [comment]
no problem, thanks for taking a look anyway :)

File: lib/main_common.pm [hunk 1]
```diff
@@ -939,9 +939,10 @@ sub load_toolchain_tests {
 }
 
 sub load_common_opensuse_sle_tests {
-    load_autoyast_clone_tests if get_var("CLONE_SYSTEM");
-    load_create_hdd_tests     if get_var("STORE_HDD_1") || get_var("PUBLISH_HDD_1");
-    load_toolchain_tests      if get_var("TCM") || check_var("ADDONS", "tcm");
+    load_autoyast_clone_tests  if get_var("CLONE_SYSTEM");
+    load_create_hdd_tests      if get_var("STORE_HDD_1") || get_var("PUBLISH_HDD_1");
+    load_toolchain_tests       if get_var("TCM") || check_var("ADDONS", "tcm");
+    loadtest "console/network" if get_var('NETWORK_CONFIGURATION');
```
Comment (okurz) [inline lib/main_common.pm]: We should try to use single ticks by default everywhere so please also in `loadtest`

File: products/sle/main.pm [hunk 2]
```diff
@@ -596,6 +596,7 @@ sub load_inst_tests {
             loadtest "installation/disk_space_fill";
         }
     }
+    loadtest "installation/network_configuration" if get_var('NETWORK_CONFIGURATION');
```
Comment (okurz) [inline products/sle/main.pm]: so why this change for SLE only?

File: tests/console/network.pm [hunk 3]
```diff
@@ -0,0 +1,28 @@
+# SUSE's openQA tests
+#
+# Copyright © 2009-2013 Bernhard M. Wiedemann
+# Copyright © 2012-2018 SUSE LLC
```
Comment (okurz) [inline tests/console/network.pm]: Just the years you worked on the file please

File: tests/console/network.pm [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Verify network configuration installed.
```
Comment (okurz) [inline tests/console/network.pm]: I don't understand this summary. Do you really verify that the network configuration is installed or something more specific?

File: tests/console/network.pm [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+use strict;
+use testapi;
+
+sub run {
+    select_console 'root-console';
+    assert_script_run 'cat /etc/sysconfig/network/dhcp | grep "DHCLIENT_SET_HOSTNAME=\"no\""';
+}
+
+sub test_flags {
+    return {milestone => 1, fatal => 1};
```
Comment (okurz) [inline tests/console/network.pm]: why?

File: tests/installation/network_configuration.pm [hunk 6]
```diff
@@ -0,0 +1,40 @@
+# SUSE's openQA tests
+#
+# Copyright © 2009-2013 Bernhard M. Wiedemann
+# Copyright © 2012-2018 SUSE LLC
```
Comment (okurz) [inline tests/installation/network_configuration.pm]: see above

File: tests/installation/network_configuration.pm [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+sub run {
+    if (!get_var("HDD_SCC_REGISTERED")) {
+        assert_screen_with_soft_timeout(
+            'scc-registration',
+            timeout      => 350,
+            soft_timeout => 300,
+            bugref       => 'bsc#1028774'
+        );
+    }
```
Comment (okurz) [inline tests/installation/network_configuration.pm]: where did you get this block from?

Comment (jknphy) [reply tests/installation/network_configuration.pm]: scc_registration.pm

Comment (okurz) [reply tests/installation/network_configuration.pm]: ok, better don't duplicate but extract method

**Comment by rwx788** [comment]
@okurz do you have any concerns? I would merge it otherwise

**Comment by jknphy** [comment]
no, I don't, anyway I will put in a development group in osd at the beginning to see how is going.

**Comment by okurz** [comment]
nicely done, thank you
